src/itest/java/uk/gov/companieshouse/exemptions/delta/exemptions/ExemptionsConsumerSteps.java

```diff
@@ -0,0 +1,200 @@
+import org.apache.kafka.clients.consumer.ConsumerRecords;
+import org.apache.kafka.clients.consumer.KafkaConsumer;
+import org.apache.kafka.clients.producer.KafkaProducer;
+import org.apache.kafka.clients.producer.ProducerRecord;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.kafka.test.EmbeddedKafkaBroker;
+import org.springframework.kafka.test.utils.KafkaTestUtils;
+import uk.gov.companieshouse.delta.ChsDelta;
+import com.github.tomakehurst.wiremock.WireMockServer;
+import uk.gov.companieshouse.exemptions.delta.util.TestDataHelper;
+
+
+public class ExemptionsConsumerSteps {
+
+    @Value("${wiremock.server.port}")
+    private String port;
+
+    private String output;
+
+    private static WireMockServer wireMockServer;
+
+    @Autowired
+    private EmbeddedKafkaBroker embeddedKafkaBroker;
+
+    @Autowired
+    private KafkaConsumer<String, byte[]> testConsumer;
+
+    @Autowired
+    private KafkaProducer<String, byte[]> testProducer;
+
+    @Autowired
+    private CountDownLatch latch;
+
+
+    @When("^the topic receives a message containing a valid CHS upsert delta payload")
+    public void consumerReceivesExemptionDeltaRequest() throws IOException, InterruptedException {
+        configureWiremock();
+        stubPutExemptions(200);
+
+        ChsDelta delta = new ChsDelta(TestDataHelper.getInputData(), 1, "123", false);
+        sendDeltaToMainTopic(delta);
+
+        assertThat(latch.await(5, TimeUnit.SECONDS)).isTrue();
+    }
+
+    @When("the consumer receives a message with invalid payload")
+    public void messageWithInvalidDataIsSent() throws Exception {
+        ChsDelta delta = new ChsDelta("invalidData", 1, "123", false);
+        sendDeltaToMainTopic(delta);
+
+        assertThat(latch.await(5, TimeUnit.SECONDS)).isTrue();
+    }
+
+    @When("an invalid avro message is sent")
+    public void invalidAvroMessageIsSent() throws Exception {
+
+        ByteArrayOutputStream outputStream = new ByteArrayOutputStream();
+        Encoder encoder = EncoderFactory.get().directBinaryEncoder(outputStream, null);
+        DatumWriter<String> writer = new ReflectDatumWriter<>(String.class);
+        writer.write("invalidData", encoder);
+        embeddedKafkaBroker.consumeFromAllEmbeddedTopics(testConsumer);
+
+        testProducer.send(new ProducerRecord<>("company-exemptions-delta", 0, System.currentTimeMillis(), "key", outputStream.toByteArray()));
+    }
+
+    @When("the topic receives a message with a valid CHS delete delta payload")
+    public void consumerReceivesExemptionsDeltaDeleteRequest() throws IOException, InterruptedException {
+        configureWiremock();
+        stubDeleteExemptions(200);
+
+        ChsDelta delta = new ChsDelta(TestDataHelper.getDeleteData(), 1, "123", true);
+        sendDeltaToMainTopic(delta);
+
+        assertThat(latch.await(5, TimeUnit.SECONDS)).isTrue();
```

Comment:
Can this be abstracted to a method to aid readability? 
Also Can the time out value be a constant so we can avoid magic numbers? This will also help in debugging where the wait can timeout while checking code at a breakpoint.
```
private static final int MESSAGE_CONSUMED_TIMEOUT = 5;

// ...

private void assertMessageConsumed() {
    assertThat(latch.await(MESSAGE_CONSUMED_TIMEOUT, TimeUnit.SECONDS)).isTrue();
}

Reply:
Done